src/exec.js

```diff
@@ -77,14 +77,14 @@ function execSync(cmd, opts, pipe) {
 
   // fs.readFileSync uses buffer encoding by default, so call
   // it without the encoding option if the encoding is 'buffer'
-  var stdout;
-  var stderr;
+  var stdout = '';
+  var stderr = '';
   if (opts.encoding === 'buffer') {
-    stdout = fs.readFileSync(stdoutFile);
-    stderr = fs.readFileSync(stderrFile);
+    try { stdout = fs.readFileSync(stdoutFile); } catch (e) {}
```

Comment:
Is this really the right fix? Why aren't the files created by the child process? I think that's where the bug really is.

Reply:
In the unit test specifically, the timeout is set to 10 ms, and node doesn't startup in that time. The code in `exec-child.js` is what is creating the stdout and stderr files, and so if node hasn't started, then the files will not exist after the `execSync` call exits.

Perhaps instead the files should be created before the `execSync` call, and the child process just writes to them?

Reply:
This sort of bug usually means that the `exec()` call inside `exec-child.js` throws an exception:

https://github.com/shelljs/shelljs/blob/8ab0a3a3931b59215553730ad86adef8b21a0fa0/src/exec-child.js#L19

Reply:
Oh, didn't see your last message. Huh, if `node` doesn't startup, then I think we have a few options:

 1. Don't try to handle this, let it be an internal error
 2. Modify the test's timeout so that we're more confident `node` can start up
 3. Create files beforehand (but this isn't really necessary in the common case, seems like a waste)

Reply:
This test case is all about checking the `timeout` option works. So we can also try something like:

```
echo start
sleep 5
echo stop # Check that this doesn't print, because we timeout before this
```

Reply:
Sorry for the delay, I'll work on this soon.

Reply:
No rush. I think it's fine to wrap this into v0.9 anyway

Reply:
The test passes when the timeout is set to 500 ms (minus the `try/catch` statements).If you think it's alright, I will just increase the timeout on the test.

Reply:
You mean to revert the try-catch statements here and to change the test timeout instead? I think that's fine.

Reply:
Ok, I'll go ahead with this.